lib/ts/fetch.ts

```diff
@@ -567,167 +566,203 @@ export async function onInvalidClaimResponse(response: ResponseWithBody) {
-        // the session is being removed. So we set it to "remove" in the
-        // cookie. This way, when we query for this token, we will return
-        // undefined (see getIdRefreshToken), and not refresh the session
-        // unnecessarily.
-
-        let expires = "Fri, 31 Dec 9999 23:59:59 GMT";
-        let cookieVal = "remove";
-        if (idRefreshToken !== "remove") {
-            let splitted = idRefreshToken.split(";");
-            cookieVal = splitted[0];
-
-            // we must always respect this expiry and not set it to infinite
-            // cause this ties into the session's lifetime. If we set this
-            // to infinite, then a session may not exist, and this will exist,
-            // then for example, if we check a session exists, and this says yes,
-            // then if we getAccessTokenPayload, that will attempt a session refresh which will fail.
-            // Another reason to respect this is that if we don't, then signOut will
-            // call the API which will return 200 (no 401 cause the API thinks no session exists),
-            // in which case, we will not end up firing the SIGN_OUT on handle event.
-            expires = new Date(Number(splitted[1])).toUTCString();
-        }
-        if (
-            domain === "localhost" ||
-            domain === WindowHandlerReference.getReferenceOrThrow().windowHandler.location.getHostName()
-        ) {
-            // since some browsers ignore cookies with domain set to localhost
-            // see https://github.com/supertokens/supertokens-website/issues/25
-            await CookieHandlerReference.getReferenceOrThrow().cookieHandler.setCookie(
-                `${ID_REFRESH_TOKEN_NAME}=${cookieVal};expires=${expires};path=/;samesite=${
-                    AuthHttpRequest.config.isInIframe ? "none;secure" : "lax"
-                }`
-            );
-        } else {
-            await CookieHandlerReference.getReferenceOrThrow().cookieHandler.setCookie(
-                `${ID_REFRESH_TOKEN_NAME}=${cookieVal};expires=${expires};domain=${domain};path=/;samesite=${
-                    AuthHttpRequest.config.isInIframe ? "none;secure" : "lax"
-                }`
-            );
+export function setToken(tokenType: TokenType, value: string, expiry: number) {
+    const name = getStorageNameForToken(tokenType);
+    // if the value of the token is "remove", it means
+    // the session is being removed. So we set it to "remove" in the
+    // cookie. This way, when we query for this token, we will return
+    // undefined (see getLocalSessionState), and not refresh the session
+    // unnecessarily.
+    logDebugMessage(`setToken: saved ${tokenType} token into cookies`);
+    return storeInCookies(name, value, expiry);
+}
+
+function storeInCookies(name: string, value: string, expiry: number) {
+    let expires = "Fri, 31 Dec 9999 23:59:59 GMT";
+    if (value !== "remove" && expiry !== Number.MAX_SAFE_INTEGER) {
```

Comment:
Maybe this function should not care about if the value is `remove`? Cause when you set front token in the FrontToken class, you call the setFrontToken function with `undefined` which just calls this with `value = ""`